### PR TITLE
CP-3694-universal-links-android

### PR DESCRIPTION
**Pull request by jprajapati1996**
- Add Deep link docs

### CODE REVIEW

File: docs/sdks/android/deeplink [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+id: extension
+title: Notification Extender Service
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: Please change the title to "Deep Links"

File: docs/sdks/android/deeplink [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+---
+
+Deep Link Android
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: Deep Links

File: docs/sdks/android/deeplink [hunk 3]
```diff
@@ -0,0 +1,63 @@
+---
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: file name: deeplinks.md

File: docs/sdks/android/deeplink [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+Deep Link Android
+--------------------
+
+1. Navigate to the app > AndroidManifest.xml and add the below code to it. As we are creating a deep link for our MainActivity.java file so we have to add this code in the MainActivity part. Below is the code which is to be added to the AndroidManifext.xml file. Comments are added in the code to get to know in more detail.
+
+2. 'AndroidManifest.xml'
+
+<intent-filter>
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: please use code block like this:
```
```java
```

File: docs/sdks/android/deeplink [hunk 5]
```diff
@@ -0,0 +1,63 @@
+---
+id: extension
+title: Notification Extender Service
+---
+
+Deep Link Android
+--------------------
+
+1. Navigate to the app > AndroidManifest.xml and add the below code to it. As we are creating a deep link for our MainActivity.java file so we have to add this code in the MainActivity part. Below is the code which is to be added to the AndroidManifext.xml file. Comments are added in the code to get to know in more detail.
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: Please optimize the grammar here a bit together with @ozamanav

File: docs/sdks/android/deeplink [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+2. 'AndroidManifest.xml'
+
+<intent-filter>
+  <action android:name="android.intent.action.VIEW" />
+  <category android:name="android.intent.category.DEFAULT" />
+  <category android:name="android.intent.category.BROWSABLE" />
+
+  <!--on below line we are specifying the host name and the scheme type in add your App name -->
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: Add space after `<!--` and grammar

File: docs/sdks/android/deeplink [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+   <data
+        android:host="https://cleverpush.com"
+        android:scheme="cleverpush" />
+</intent-filter>
+  
+<!--below is the same filter as above just the scheme is changed to your app name -->
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: Add space after `<!--` and grammar

File: docs/sdks/android/deeplink [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  <data
+    android:host="www.cleverpush.com"
+    android:scheme="cleverpush" />
+</intent-filter>
+
+3. Working with the MainActivity.java file
+
+Go to the MainActivity.java file and refer to the following code. Below is the code for the MainActivity.java file. Comments are added inside the code to understand the code in more detail.
+
+public class MainActivity extends AppCompatActivity {
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: code block

File: docs/sdks/android/deeplink [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  <category android:name="android.intent.category.BROWSABLE" />
+  
+  <data
+    android:host="www.cleverpush.com"
+    android:scheme="cleverpush" />
+</intent-filter>
+
+3. Working with the MainActivity.java file
+
+Go to the MainActivity.java file and refer to the following code. Below is the code for the MainActivity.java file. Comments are added inside the code to understand the code in more detail.
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: grammar

File: docs/sdks/android/deeplink [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+		
+		// after that we are extracting string from that parameters.
+		String param = parameters.get(parameters.size() - 1);
+		
+		//You can check in our Logcat in your param parameter Log.
+		Log.e("param",param);
+    
+	  }
+  }
+}
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: add empty line at end of file

File: docs/sdks/android/deeplink [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+	// checking if the uri is null or not.
+	if (uri != null) {
+		// if the uri is not null then we are getting the
+		// path segments and storing it in list.
+		List<String> parameters = uri.getPathSegments();
+		
+		// after that we are extracting string from that parameters.
+		String param = parameters.get(parameters.size() - 1);
+		
+		//You can check in our Logcat in your param parameter Log.
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: space after //

File: docs/sdks/android/deeplink [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+</intent-filter>
+  
+<!--below is the same filter as above just the scheme is changed to your app name -->
+<intent-filter>
+  <action android:name="android.intent.action.VIEW" />
+  <category android:name="android.intent.category.DEFAULT" />
+  <category android:name="android.intent.category.BROWSABLE" />
+  
+  <data
+    android:host="www.cleverpush.com"
```
Comment (mariusglauer) [inline docs/sdks/android/deeplink]: Customer needs to change host= right? so we should add comment here

**Review by mariusglauer — CHANGES_REQUESTED** [review]

File: docs/sdks/android/deeplink (hunk 1 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: Resolved

File: docs/sdks/android/deeplink (hunk 2 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: Resolved

File: docs/sdks/android/deeplink (hunk 3 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: Resolved

File: docs/sdks/android/deeplink (hunk 6 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: Resolved

File: docs/sdks/android/deeplink (hunk 5 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: Navigate to app > AndroidManifest.xml and add the below code to it. As we are creating a deep link for our MainActivity.java file so we have to add this code in the MainActivity part.Comments are added in the code to get to know in more detail.


Is it okay...?

File: docs/sdks/android/deeplink (hunk 7 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: Resolved

File: docs/sdks/android/deeplink (hunk 9 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: Resolved

File: docs/sdks/android/deeplink (hunk 10 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: Added

File: docs/sdks/android/deeplink (hunk 11 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: Added Space

File: docs/sdks/android/deeplink (hunk 12 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: <data
    android:host="www.cleverpush.com"
    android:scheme="cleverpush" />


Is it okay..??

File: docs/sdks/android/deeplink (hunk 4 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: Resolved

File: docs/sdks/android/deeplink (hunk 8 above)
Comment (jprajapati1996) [reply docs/sdks/android/deeplink]: Resolved

**Pull request merged**